packages/desktop-client/src/components/reports/reports/Spending.tsx

```diff
@@ -47,25 +47,54 @@ export function Spending() {
       conditions: filters,
       conditionsOp,
       setDataCheck,
+      mode,
     });
-  }, [categories, filters, conditionsOp]);
+  }, [categories, filters, conditionsOp, mode]);
 
   const data = useReport('default', getGraphData);
   const navigate = useNavigate();
   const { isNarrowWidth } = useResponsive();
 
-  if (!data) {
-    return null;
-  }
+  // Move the conditional logic outside of the hooks
   const showAverage =
+    data &&
     data.intervalData[27].months[
       monthUtils.subMonths(monthUtils.currentDay(), 3)
     ].daily !== 0;
+
   const todayDay =
     monthUtils.getDay(monthUtils.currentDay()) - 1 >= 28
       ? 27
       : monthUtils.getDay(monthUtils.currentDay()) - 1;
 
+  const showLastYear = data && Math.abs(data.intervalData[27].lastYear) > 0;
+  const showLastMonth = data && Math.abs(data.intervalData[27].lastMonth) > 0;
+  const showGraph = showLastMonth || showLastYear || showAverage;
+  useEffect(() => {
+    if (mode === 'Last month' && !showLastMonth) {
```

Comment:
I'm not sure these will ever be used since we are hiding the buttons (below) it's not possible to select a button that doesn't have data. The default "this month" is caught by your `showGraph` message (below).

Reply:
It's actually required. Sent you a discord message explaining why. If you have a better idea how this could be achieved, please let me know :)